redbot/core/commands/help.py

```diff
@@ -274,7 +274,7 @@ async def send_help(
     async def get_cog_help_mapping(
         self, ctx: Context, obj: commands.Cog, help_settings: HelpSettings
     ):
-        iterator = filter(lambda c: c.parent is None and c.cog is obj, ctx.bot.commands)
+        iterator = obj.get_commands()
```

Comment:
Fails with `AttributeError: 'NoneType' object has no attribute 'get_commands'` when fetching the entire help output (`[p]help` with no arguments). 

```py
Traceback (most recent call last):
  File "c:\users\flame\venv\dev\lib\site-packages\discord\ext\commands\core.py", line 229, in wrapped
    ret = await coro(*args, **kwargs)
  File "c:\users\flame\venv\dev\lib\site-packages\redbot\core\commands\help.py", line 911, in red_help
    await ctx.bot.send_help_for(ctx, thing_to_get_help_for, from_help_command=True)
  File "c:\users\flame\venv\dev\lib\site-packages\redbot\core\bot.py", line 1272, in send_help_for
    return await self._help_formatter.send_help(
  File "c:\users\flame\venv\dev\lib\site-packages\redbot\core\commands\help.py", line 252, in send_help
    await self.format_bot_help(ctx, help_settings=help_settings)
  File "c:\users\flame\venv\dev\lib\site-packages\redbot\core\commands\help.py", line 631, in format_bot_help
    coms = await self.get_bot_help_mapping(ctx, help_settings=help_settings)
  File "c:\users\flame\venv\dev\lib\site-packages\redbot\core\commands\help.py", line 296, in get_bot_help_mapping
    cm = await self.get_cog_help_mapping(ctx, cog, help_settings=help_settings)
  File "c:\users\flame\venv\dev\lib\site-packages\redbot\core\commands\help.py", line 277, in get_cog_help_mapping
    iterator = obj.get_commands()
AttributeError: 'NoneType' object has no attribute 'get_commands'
```

Reply:
This happens specifically because `None` is passed to this method as a "cog" object to catch any commands with no cog associated with them when listing all commands in the bot. I don't know if there's any special way to get a list of commands *not* associated with a cog, so this might need to use the existing check in this case.
https://github.com/Cog-Creators/Red-DiscordBot/blob/aa51fd9ad1463484039f1578ae71210a516000b8/redbot/core/commands/help.py#L295-L296